plugins/modules/keycloak_group.py

```diff
@@ -69,6 +69,18 @@
             - A dict of key/value pairs to set as custom attributes for the group.
             - Values may be single values (e.g. a string) or a list of strings.
 
+    parents:
+        version_added: "6.3.0"
+        type: list
+        description:
+            - List of parent groups for the group to handle sorted top to bottom.
+            - Set this to create a group as a subgroup of another group or groups (parents).
+            - Each element can describe a parent either by name or ID, both is
+              fine but using names needs some more internal API calls (to map
+              to ID's internally). On default given strings are interpreted as
+              names, to mark them as ID's prefix them with C(id:)
```

Comment:
```suggestion
              names, to mark them as ID's prefix them with C(id:).
```
And if you ever try to name one of your groups `id:software` and want to create subgroups in it, you shall burn in hell ;-)

Reply:
Fixed the missing period. In principle you actually might have valid a point with the restricting vocabulary thing, yeah prefixing a group names with `id:` might sound like a fringe case but artificially limiting the expression power of a system feels not right. The more I thing about the less I like it. I mean there are potential alternative solutions to this:

1. Dont use any magic extra syntax and try to identify ID's per regex:
    - Restricts group naming in another way (names cannot match id regex)
    - Using regexes makes this implementation more brittle and the ansible module must also be aware of ID's structure which ideally it shouldn't matter here

2. Actually having two parameters here instead of one for parents, something like `parent_ids` and `parent_names` similar like the module has two different parameters for group name and group id:
    - Having two parameters make internal and external handling of the module interface more complicated
    - It is not possible anymore to have a mixed parent resolution path with some given by names and others by ID's, which admittedly might be a rather edge case thing anyway
    
3. Use subdicts instead of string for parent chain, it could look something like this:
    ```
      parents:
         - type: id
           value: {{ pgroup.id }}
         - type: name
           value: "foo"
    ```
     
    - this should work and is rather clean but more typing work for the probably "90%" simple cases which are non conflicting

4. It would also be possible to add an additional optional `name:` prefix analogous to `id:` to use as escape sequence, this should also be fine:
    ```
    foo => handled as name "foo"
    id:... => handled as id
    name:id:foo => handled as name "id:foo"
    name:name:foo =>handled as name "name:foo"
    ```

From the four I could think of I currently would prever the last one, any thoughts?

Reply:
You could also use subdicts as follows:
```
  parents:
     - id: "{{ pgroup.id }}"
     - name: "foo"
```
i.e. allow two entries `id` and `name` that are mutually exclusive and one of them is required.

I prefer explicit forms like this or 3. over string parsing forms like 4 and the current version. (I would use string parsing forms only when the system/software/... the module is for is using that same notation.)

Reply:
Yeah, you are probably right, using a more structured approach is always the better idea. So I overhauled the parameter part one more time and adopted your suggestion. Quite liking the final result, the look and feel is very similar to the previous simple string solution but at the same time it is structurally much better.